src/theory/uf/eq_proof.h

```diff
@@ -35,7 +35,9 @@ namespace eq {
 class EqProof
 {
  public:
-  EqProof() : d_id(MERGED_THROUGH_REFLEXIVITY) {}
+  EqProof(NodeManager* nm) : d_nm(nm), d_id(MERGED_THROUGH_REFLEXIVITY) {}
```

Comment:
Can you try to avoid adding this?  I'd prefer not adding it to this class since it is very low level.

I'm curious why this is necessary?

Reply:
This is necessary because there are several calls to `mkConst` and `mkNode` that take a vector as an argument, namely:

- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R527
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R597
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R954
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R959
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1006
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1087
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1121
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1320
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1321
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1335
- https://github.com/cvc5/cvc5/pull/11423/files#diff-a6640faf8dda114c4706f1b6e1921d1fa84e4908add6b77f2ef1fe038d690e63R1336

My understanding is that we preferred storing a pointer to the NodeManager in the class rather than retrieving it from local nodes within different functions.

Reply:
In this case, I'd rather pass the node manager to the methods.

This is because `EqProof` is a data structure (similar to Node), not a module.